daemon/info.go

```diff
@@ -64,9 +64,9 @@ func (daemon *Daemon) SystemInfo() *types.Info {
 		Labels:             daemon.configStore.Labels,
 		ExperimentalBuild:  daemon.configStore.Experimental,
 		ServerVersion:      dockerversion.Version,
-		HTTPProxy:          maskCredentials(getEnvAny("HTTP_PROXY", "http_proxy")),
-		HTTPSProxy:         maskCredentials(getEnvAny("HTTPS_PROXY", "https_proxy")),
-		NoProxy:            getEnvAny("NO_PROXY", "no_proxy"),
+		HTTPProxy:          maskCredentials(getEnvOrConfig(daemon.configStore.HttpProxy, "HTTP_PROXY", "http_proxy")),
+		HTTPSProxy:         maskCredentials(getEnvOrConfig(daemon.configStore.HttpsProxy, "HTTPS_PROXY", "https_proxy")),
+		NoProxy:            getEnvOrConfig(daemon.configStore.NoProxy, "NO_PROXY", "no_proxy"),
```

Comment:
Hm.. but this only changes what's shown in `docker info`, but not actually configures the daemon to use these proxies? 🤔 

FWIW, default behavior; https://github.com/golang/go/blob/912f0750472dd4f674b69ca1616bfaf377af1805/src/net/http/transport.go#L422-L441 / https://github.com/golang/go/blob/912f0750472dd4f674b69ca1616bfaf377af1805/src/net/http/transport.go#L43-L54

(and we are using https://github.com/docker/go-connections for some central/default configuration, which may be relevant)